packages/terra-application/src/page/PageHeader.jsx

```diff
@@ -0,0 +1,220 @@
+   * A NotificationBanners element type (as provided by the useNotificationBanners
+   * hook) to be rendered alongside the other header contents.
+   */
+  NotificationBanners: PropTypes.elementType,
+  /**
+   * A function to be executed upon selection of the Page's back button. If not
+   * provided, the back button will not be rendered.
+   */
+  onSelectBack: PropTypes.func,
+  /**
+   * A PageToolbar element to render alongside the other header contents.
+   */
+  toolbar: PropTypes.element,
+};
+
+const PageHeader = ({
+  actions, id, label, NotificationBanners, onSelectBack, toolbar,
+}) => {
+  const intl = React.useContext(ApplicationIntlContext);
+  const pageContainerContext = React.useContext(PageContainerContext);
+
+  const headerContainerRef = React.useRef();
+  const moreActionsButtonRef = React.useRef();
+
+  const [showMenu, setShowMenu] = useTransientPresentationState(false);
+  const { activeBreakpoint } = useElementSize(headerContainerRef, breakpointFilter);
+
+  // We filter out any falsy action values that may have been provided
+  // due to inline boolean logic.
+  let validActions = [];
+  if (actions) {
+    validActions = React.Children.toArray(actions.props.children).filter(child => child);
+  }
+
+  if (validActions.length > 3) {
+    throw new Error(`[terra-application] ${label} cannot render more than three actions.`);
+  }
+
+  const actionsAreCollapsed = activeBreakpoint === 'tiny' && validActions.length > 1;
+
+  function renderActionButtons() {
+    return validActions.map((action) => (
+      <PageHeaderButton
+        key={action.key}
+        refCallback={action.props.refCallback}
+        icon={action.props.icon}
+        ariaLabel={action.props.label}
+        onSelect={action.props.onSelect}
+      />
+    ));
+  }
+
+  function renderMenuButton() {
+    return (
+      <PageHeaderButton
+        refCallback={(ref) => {
+          moreActionsButtonRef.current = ref;
+
+          validActions.forEach((action) => {
+            if (action.props.refCallback) {
+              action.props.refCallback(ref);
+            }
+          });
+        }}
+        className={cx('header-button')}
+        icon={<IconRollup />}
+        ariaLabel={intl.formatMessage({
+          id: 'terraApplication.pageHeader.moreActions',
+        })}
+        onSelect={() => {
+          setShowMenu(true);
+        }}
+      />
+    );
+  }
+
+  function renderMenu() {
+    return (
+      <Popup
+        isOpen={showMenu}
+        targetRef={() => moreActionsButtonRef.current}
+        onRequestClose={() => { setShowMenu(false); }}
+        contentHeight="auto"
+        contentWidth="auto"
+        contentAttachment="top right"
+        isContentFocusDisabled
+        isHeaderDisabled
+        popupContentRole="none"
+      >
+        <ActionMenu
+          label={intl.formatMessage({
+            id: 'terraApplication.pageHeader.actionsMenu',
+          }, { label })}
+          onRequestClose={() => { setShowMenu(false); }}
```

Comment:
if you update above, let's do the same here for consistency
```suggestion
          onRequestClose={() => setShowMenu(false)}
```